crates/node/p2p/src/gossip/gater.rs

```diff
@@ -30,6 +31,9 @@ pub struct ConnectionGater {
     pub blocked_peers: HashSet<PeerId>,
     /// A set of blocked ip addresses that cannot be dialed.
     pub blocked_addrs: HashSet<IpAddr>,
+
+    /// A set of blocked subnets that cannot be connected to.
+    pub blocked_subnets: HashSet<String>,
```

Comment:
Can we change this to a `HashSet<IpNet>`?

This way, every time `check_ip_in_blocked_subnets` is called, the subnet string doesn't need to be parsed.